src/os/bluestore/BlueStore.cc

```diff
@@ -1641,6 +1639,23 @@ void BlueStore::SharedBlob::put_ref(uint64_t offset, uint32_t length,
     unshare && !*unshare ? unshare : nullptr);
 }
 
+void BlueStore::SharedBlob::finish_write(uint64_t seq)
+{
+  while (true) {
+    Cache *cache = coll->cache;
```

Comment:
Wondering why _osr_drain_preceding stuff from _split_collection doesn't protect against this issue?

Reply:
Oh.. yeah.  The problem seems to be that the OSD submits the split txn on the parent pg sequencer, while store_test does it on the child pg sequencer.  And we haven't noticed because there's no deferred io generated by that test.  I'll change that as well.

I'm inclined to keep the retry in place, though...